### ISSUE TITLE
JRuby support

### ISSUE THREAD

**Issue by inawlaljar** [doorkeeper-gem/doorkeeper#612]
When I run the rails server on a development machine, it throws the following error. 

```
/Users/Raj/.rvm/gems/jruby-1.7.19/gems/doorkeeper-2.1.4/lib/doorkeeper/engine.rb:4: unknown type of %string
      app.config.filter_parameters += %i(client_secret code token)
```

I am using doopkeeper with jruby 1.7.19 & rails 4.2. 

Cheers,
Raj

**Comment by tute** [comment]
Hi @inawlaljar. Thanks for reporting. `%i(` in MRI creates an array of symbols. I never thought of its implications in JRuby. Would you send a PR adding `- jruby-head` to the `.travis.yml` file, so we add JRuby support to doorkeeper?

Thanks again!

**Comment by jonahb** [comment]
The problem isn't lack of JRuby support per se but that %i was introduced in Ruby 2, and JRuby 1.7.x is compatible with Ruby 1.9.3. The feature might be better named "Ruby 1.9 support." (Hi @tute!)

**Comment by tute** [comment]
Hey @jonahb! Ruby 1.9.3 has reached end of life though, the reason why I prefer doorkeeper not to support it. If they work the same the names are interchangeable indeed; if they don't I will favor JRuby.
Thank you for your input!

**Comment by jonahb** [comment]
Oh, I wasn't suggesting that you support 1.9.3, just pointing out that OP's problem is not related to JRuby in particular. It would occur on any 1.9.3-compatible interpreter. Even if the tests pass on jruby-head, OP's problem will still exist; he'll have to upgrade his JRuby.

**Comment by tute** [comment]
@inawlaljar are you still interested in this? Would you like to work on it?

**Comment by jasl** [comment]
@tute let's we add JRuby to Travis-CI and tolerate it failure, maybe my team will use JRuby in future and I can working on this(but I can't make a promise).

**Comment by nonnenmacher** [comment]
fall in same trap today, after update to latest version.

On my ruby 1.9.3 test machine (used solely to test my app still run in ruby 1.9.3)
I fall into this line.

A simple

           app.config.filter_parameters += [:client_secret, :code, :token]

make it work again.

I don't think this is a bug deal, until a precise drop-out of 1.9.3 is announced in advance, as of 2.2.0 seems to work fine, with just this Ruby 2 `%i` nicety kept away for the moment.

**Comment by tute** [comment]
Closing in favor of PR.

**Comment by tute** [comment]
I can't find why JRuby doesn't work well in CI in #638, and I don't have time to investigate it. If you want to build upon that PR you are welcome. If you want to unroll in another PR the usage of `%i` that's fine with me (but then I couldn't guarantee other pieces of the library are JRuby compatible).

Thanks!

**Comment by tute** [comment]
Released 2.2.1 with this fix. Thanks!

**Comment by tute** [comment]
Jruby testing is in master. :heart:
